File: Data Structures & Algorithms/depth-of-binary-tree/submission-3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0

        maxDepth = self.dfs(root, 1)

        return maxDepth

    def dfs(self, node, curDepth):
        if not node:
            return curDepth

        leftMax, rightMax = curDepth, curDepth
        if node.left:
            leftMax = self.dfs(node.left, curDepth + 1)
        if node.right:
            rightMax = self.dfs(node.right, curDepth + 1)

        return max(leftMax, rightMax)

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-3.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        var left = 0;
        var right = 1;

        var currMax = 0;
        while (right < prices.Length)
        {
            if (prices[left] > prices[right])
            {
                left = right;
                right++;
                continue;
            }

            var profit = prices[right] - prices[left];
            currMax = Math.Max(currMax, profit);
            right++;
        }

        return currMax;
    }
}
